### ISSUE TITLE
Cameras

### ISSUE THREAD

**Issue by UKenGB** [Samfox2/homebridge-videodoorbell#74]
How do multiple cameras work?
Do they appear separately in Home.app?
Does button push event Notification end snapshot of only the camera with port specified in HTTP POST?
Can the same Platform be used with multiple cameras, one at each of 2 different doors?
Would it be possible to have a 'Camera' with no actual camera (no videoConfig section?), so just use with a button that triggers the notification?

**Comment by Samfox2** [comment]
They appear as different cameras and triggered with their specific port. One platform can handle multiple cameras. As this is a videodoorbell-plugin a camera should be assigned. However you can put anything in the source part and do only use the trigger but the cam will show up in home app with a black image.

**Comment by UKenGB** [comment]
That's what I thought. Thing is, I added another camera with 'sources' that all work as they should and other parameters same or similar to the first one, but now nothing works. The config is all good and saves without error and I've restarted Homebridge several times, but testing by sending 'curl…' with either of the specified ports produces no Notification at all. 
I'll cut the extra camera and see if it works again, but at the moment it looks like more than one camera stops it working.

**Comment by UKenGB** [comment]
Cutting the new camera didn't help and the original cam no longer works.
I changed some names, are they sync'd with HomeKit so no longer linked?
I'll check in Home.app.

**Comment by UKenGB** [comment]
Yup, Home.app cannot find them. Name change completely confused it. Shame there's no way to simply update Home.app for new names.

Removed and re-added them and now I am getting appropriate Notifications for both cams.

**Comment by Samfox2** [comment]
Yes sometimes those syncing issues between icloud and ios devices are really a pain.

**Comment by UKenGB** [comment]
> …As this is a videodoorbell-plugin a camera should be assigned. However you can put anything in the source part and do only use the trigger but the cam will show up in home app with a black image.

Would I be correct in thinking that although a missing 'source' line prevents Homebridge from even starting, 'stillImageSource' is not actually required?

An actual question though, should I not be able to point "stillImageSource" to an image on the Homebridge server and have that appear in Home.app? I tried that with the absolute path to an image on that server, but it never appeared in Home.app. There is an issue with Apple's app anyway as it keeps using the last image it found, even though Homebridge's config now points to a totally different image. Home.app needs a 'Refresh' button.

But back to using a static image, should I be able to point to an image file located on the server and in which case, what would be the correct path?

**Comment by Samfox2** [comment]
Yes, still image part is not required. It should also work with static images. If you have any issues with that, please try to check the resolution of the image first. It has to be supported by homekit. If it is still not working, please make sure that both image path and access rights are ok. Homebridge user needs to be allowed to read the image.

**Comment by UKenGB** [comment]
Ah thanks, didn't realise HomeKit had specific resolution requirements. Can you let me know what they are, or point me to this info somewhere?

I'll check perms again, but should be good to read.

**Comment by Samfox2** [comment]
You can see them in file `ffmpeg.js` at the beginning where resolutions are checked.

**Comment by UKenGB** [comment]
I converted the image to 480x360 and that seemed to do the trick. Of course, I really should buy another camera.

I was looking in ffmpeg.js and cannot understand the 'if, else' layout to determine width and height. It looks to me like it always has to check every condition, even if a previous one has matched. That’s a terrible way to do it and makes me think I’m missing something about what it’s doing. Could you help me understand what it is trying to do?

BTW, which ffmpeg is used by default? I have an up-to-date one installed in the PATH, but does this plug-in use its own included version? I had a look but couldn’t find anything contained in the plug-in so assume it uses the system’s default one. Is that correct?

Happy New Year in any case. :-)


Ken  G i l l e t t

_/_/_/_/_/_/_/_/

**Comment by Samfox2** [comment]
It simply checks which maximum compatible resolution is supported an can be used.

**Comment by Bibkins** [comment]
If you think about the security of the house, office, or industrial premises, you thought about the cameras, and rightly so, for example, I installed a [best wired camera security system](https://gohomesystems.com/best-wired-security-camera-system/) around the perimeter of the house and in the house. I do not regret my decision. I advise you to contact this company to get cameras, so you can sleep well.

**Comment by Dovofeev55** [comment]
https://crftvideo.com/
